builtin/credential/github/path_login.go

```diff
@@ -203,16 +217,31 @@ func (b *backend) verifyCredentials(ctx context.Context, req *logical.Request, t
 		orgOpt.Page = resp.NextPage
 	}
 
+	orgLoginName := ""
 	for _, o := range allOrgs {
-		if strings.EqualFold(*o.Login, config.Organization) {
+		if *o.ID == config.OrganizationID {
 			org = o
+			orgLoginName = *o.Login
 			break
 		}
 	}
 	if org == nil {
 		return nil, logical.ErrorResponse("user is not part of required org"), nil
 	}
 
+	if orgLoginName != config.Organization {
+		// the org name has changed, update it
+		config.Organization = orgLoginName
+		entry, err := logical.StorageEntryJSON("config", config)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		if err := req.Storage.Put(ctx, entry); err != nil {
```

Comment:
If this storage.Put is necessary, we might want to perform a single write at the end instead of two separate writes within this func to avoid partial writes on failures.